doc/security/specs/secure_boot/index.md

```diff
@@ -6,22 +6,96 @@ title: "Secure Boot"
 
-# References
+The diagram below summarizes the specific steps involved in the secure boot process:
 
-*   [Attestation][attestation]
-*   [Ownership Transfer][ownership-transfer]
-*   [Memory Protection][mask-rom-epmp]
-*   [OpenTitan Flash][ot-flash]
+<img src="secure_boot_flow.svg" style="width: 800px;">
 
-# Terminology
+## Mask ROM {#mask-rom}
+
+The first stage of secure boot is called "Mask ROM".
+Mask ROM is a region of read-only memory that cannot be updated at all after an OpenTitan device is manufactured.
+For that reason, the Mask ROM is kept as simple as possible; it does some minimal setup, authenticates the next stage (`ROM_EXT`), and jumps there.
+
+The Mask ROM contains non-updateable public keys, which it uses to authenticate the `ROM_EXT` signature.
+These public keys correspond to what OpenTitan calls the "Silicon Creator": the entity who was initially involved in the manufacturing of the device.
+It's important to distinguish between the unchanging Silicon Creator and the changeable "Silicon Owner", the entity that owns the device at a given time.
+
+On startup, hardware settings use a feature called [“enhanced Physical Memory Protection” (ePMP)][ibex-epmp] to ensure that only the Mask ROM code itself is executable.
+Flash and MMIO memory regions are unlocked for read/write (for signature verification and configuring peripherals) but are not executable at this stage.
+
+On boot, the Mask ROM code does the following:
+1. Initialize any additional ePMP settings beyond what is set by hardware at startup, plus initialize the C runtime and hardware peripherals (e.g. pinmux, UART).
+    * Load the "manifest" for `ROM_EXT`, which includes the start and end address of `ROM_EXT` code, a cryptographic signature, a public key modulus, and “selector bits” for hardware information.
+    * The manifest's modulus field identifies which of the stored [Silicon Creator keys][silicon-creator-keys] should be used for signature verification. If there is no matching stored key, or the matching key is not suitable for the device’s lifecycle state, then the boot fails.
+2. Read usage constraints from hardware according to the selector bits from the manifest.
+    * The selector bits can choose all or part of the device ID, and/or information about the state of the device. Because the selected constraints are included in the digest, this means the signer of a `ROM_EXT` image can restrict their signature to only certain devices/states.
+3. Compute the SHA2-256 digest of the selected usage constraints concatenated with the contents of memory between the `ROM_EXT` start and end address given in the manifest.
+4. Check the signature from the manifest against the digest and the selected Silicon Creator public key.
+    * If this step fails, it’s retried just in case of a transient error, but only once. If the second try fails, the boot fails.
```

Comment:
nit: we retry with the second slot, not the same slot.

should this be mentioned somewhere earlier?

Reply:
Ah, good point, I'll clarify earlier that there are two `ROM_EXT` slots and that the second try is with the other slot. How does the boot procedure decide which slot to try first? Is it part of the boot policy page or is it chosen somewhere else?

Reply:
We pick the one with the newer security version ([here](https://github.com/lowRISC/opentitan/blob/5bb79157b9dea77f485f869938b8ddaa8ec93b59/sw/device/silicon_creator/mask_rom/boot_policy.c#L14)).